### DISCUSSION TITLE
Launch.json configuration to debug code server

### DISCUSSION THREAD

**Discussion by ericzhucode** [coder/code-server#4888] [Q&A]
Hi team, I have read contribution documents you published
https://coder.com/docs/code-server/latest/CONTRIBUTING
Everything is perfect but may I know how your team debug this project?
If we use vscode or code server to debug, then we have to set **launch.json** configurations.
In vscode, official configurations is offered and we can debug all processes directly but in code server, there's no such file yet.
I have wrote my own launch.json and I can debug express framework of code server and render process of vscode now, but I found it's still hard to debug other process like extension host process.
Can you let us community contributors to know the way you debug this project? Or is it possible that you can add those settings file to source code like vscode does?
Just a suggestion, thank you so much for your effort so far.

**Comment by lg-nineoneninethreenine** [comment]
We really rely on the console for this. We execute watching mode and other commands, so it hot reloas with error logs.

**Reply by jsjoeio** [reply]
Echoing @im-coder-lg, I usually debug things with console logs. So assuming you've followed the CONTRIBUTING docs, then I usually drop logs into the place where I need to debug something then look for those in the server or the browser. 

I hope that helps a little bit! If not, let us know what else would be helpful!

**Comment by lg-nineoneninethreenine** [comment]
What do you actually mean? I don't think we need a launch.json for code-server.
